### ISSUE TITLE
Issue installing with 0.66.0

### ISSUE THREAD

**Issue by gakinson** [launchdarkly/react-native-client-sdk#103]
**Is this a support request?**
This issue tracker is maintained by LaunchDarkly SDK developers and is intended for feedback on the SDK code. If you're not sure whether the problem you are having is specifically related to the SDK, or to the LaunchDarkly service overall, it may be more appropriate to contact the LaunchDarkly support team; they can help to investigate the problem and will consult the SDK team if necessary. You can submit a support request by going [here](https://support.launchdarkly.com/hc/en-us/requests/new) or by emailing support@launchdarkly.com.

Note that issues filed on this issue tracker are publicly accessible. Do not provide any private account information on your issues. If your problem is specific to your account, you should submit a support request as described above.

**Describe the bug**

Trying to install the react native package while using react native 0.66.0 results in the below. I tried doing a force install and then when I try to run, I get the error 

```
fatal error: module map file '/Users/geoffreykinson/Library/Developer/Xcode/DerivedData/PROJECT/Build/Products/Debug-iphoneos/LDSwiftEventSource/LDSwiftEventSource.modulemap' not found
```

```
code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: wesponsoredapp@0.0.1
npm ERR! Found: react-native@0.66.0
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.66.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react-native@"~0.63 || ~0.64" from launchdarkly-react-native-client-sdk@5.1.0
npm ERR! node_modules/launchdarkly-react-native-client-sdk
npm ERR!   launchdarkly-react-native-client-sdk@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/geoffreykinson/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/geoffreykinson/.npm/_logs/2021-12-03T20_35_17_478Z-debug.log
geoffreykinson@geoffreys-mbp WeSponsoredApp % cd ios/
geoffreykinson@geoffreys-mbp ios % npm install --save launchdarkly-react-native-client-sdk --force
npm WARN using --force Recommended protections disabled.
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: react-native@0.66.0
npm WARN node_modules/react-native
npm WARN   react-native@"0.66.0" from the root project
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react-native@"~0.63 || ~0.64" from launchdarkly-react-native-client-sdk@5.1.0
npm WARN node_modules/launchdarkly-react-native-client-sdk
npm WARN   launchdarkly-react-native-client-sdk@"*" from the root project

added 1 package, removed 8 packages, changed 1 package, and audited 1844 packages in 4s

```

**To reproduce**
```
npm install --save launchdarkly-react-native-client-sdk
```

**Expected behavior**
Package install

**Logs**
If applicable, add any log output related to your problem.

**SDK version**
The version of this SDK that you are using.

**Language version, developer tools**
For instance, Go 1.11 or Ruby 2.5.3. If you are using a language that requires a separate compiler, such as C, please include the name and version of the compiler too.

**OS/platform**
Mac

**Additional context**

**Comment by louis-launchdarkly** [comment]
Hello @gakinson, thank you for reaching out. As stated in [Supported Versions](https://github.com/launchdarkly/react-native-client-sdk#supported-versions), we are supporting up to React Native 0.64.x - We are currently working on supporting the newer version. I will mark this as an enhancement request.

Filed Internally as 133606.

**Comment by gakinson** [comment]
> .

Thanks for the response. Do you have an ETA on this? It's fairly important for our app launch to have launch darkly integrated and downgrading has a higher risk ATM

**Comment by gakinson** [comment]
Following up on question

**Comment by gakinson** [comment]
Following up on this

**Comment by louis-launchdarkly** [comment]
Hello @gakinson, apologized for the delayed response. Unfortunately, we are still resolving issues with the 0.65.x and 0.66.x build internally and do not have an ETA for it yet.

**Comment by gauthierm** [comment]
What are the issues? Can you log them here in the event that someone in the community knows the solution?

**Comment by gauthierm** [comment]
@louis-launchdarkly Ping, can you share more details about the issues?

**Comment by sebasthuis** [comment]
We have currently upgraded our RN project to 0.65.0, but have not experienced any problems with LaunchDarkly. It seems that everything is working fine currently. Having some additional details would help me understand what the ramifications are of releasing this.

**Comment by finalight** [comment]
may I know is the 0.65.x officially supported?

**Comment by louis-launchdarkly** [comment]
Hello @gauthierm, from my discussion with @gwhelanLD, our design choice of wrapping LaunchDarkly native Android and iOS SDKs leads to a lot of complexity in the build and validation to ensure reliable SDK releases. We are fixing an issue with a dependency on one of the native SDKs.

Hello @sebasthuis and @finalight - We are actively working on the RN 0.65 and RN 0.66 build, while I don't have an exact date yet, I am hoping to have an update that unblocks you soon, as we know that many people is reporting and clamoring for the RN 0.65 and RN 0.66 support.

**Comment by gakinson** [comment]
Hello @louis-launchdarkly,

I was wondering if the javascript sdk will work for react native? Thinking of work arounds in the meantime.

Thank you,
Geoffrey Kinson

**Comment by samih7** [comment]
@gakinson FYI, LaunchDarkly `4.2.2` with React Native `0.67.2` seems to be working as expected so far.

**Comment by EdwardDrapkin** [comment]
Version 68 is about to be out.

Why are you not supporting versions that have been EOL'd yet?

This is making your service an exceptionally hard sell for me.  I can't downgrade react-native and I flat cannot find an SDK that works on React Native 66.  We're likely going to sign a six/seven figure contract with somebody, should I get a sales person to light a fire under y'all?  What is taking so long?

**Comment by finalight** [comment]
is it working with 0.67? i think someone mention about it working with 0.67

**Comment by samih7** [comment]
@finalight @gakinson Yes, for us it works with React Native `0.67.2` (LaunchDarkly `4.2.2`).

**Comment by wojteg1337** [comment]
I'm not a someone!

**Comment by gakinson** [comment]
> @finalight @gakinson Yes, for us it works with React Native `0.67.2` (LaunchDarkly `4.2.2`).

Thank you 🙏
